src/ts/ConvivaAnalytics.ts

```diff
@@ -465,7 +498,11 @@ export class ConvivaAnalytics {
 
     if (playerState) {
       this.debugLog('[ ConvivaAnalytics ] report playback state', playerState);
-      this.playerStateManager.setPlayerState(playerState);
+      if (this.adTrackingModule && this.adTrackingModule.isAdSessionActive()) {
```

Comment:
how can there be no `adTrackingModule`? Shouldn't we check for `isAdActive`

Reply:
This is to be bullet prove. In general your statement is correct.

But we bind the player events in the constructor but `this.adTrackingModule` will only be set when the session get initialized. So it could happen that this method is called before we have an active session.

-> No we can't initialize the `this.adTrackingModule` in the constructor because it needs an active tracking session.